### DISCUSSION TITLE
How should I customize the field values displayed on the front-end page？

### DISCUSSION THREAD

**Discussion by bibiHY** [allure-framework/allure2#2106] [Questions & Support] [answered]
Hello, I am currently using testNG with the allure. The field content displayed in the generated test report is fixed. Can I customize a certain field to display the content I want to display?

For example, I would like to replace the value in arg1 with a value in the image that reads “已完成订单申请退款,形成售后单1”. As shown in the figure
![image](https://github.com/allure-framework/allure2/assets/99535061/de5cfca3-237a-4fc7-9f70-20ffe952b611)

**Comment by baev** [comment, marked as answer]
There are two options available: 

1. Use `-parameters` compiler option. That way, the real parameter names will be used instead of `argN`
2. Use `io.qameta.allure.Param` annotation:
  ```java
  void testSomething(@Param("Human-readable parameter name") String param1) {
      // ...
  }
  ```
